synfig-studio/src/gui/widgets/widget_link.cpp

```diff
@@ -49,61 +52,141 @@ using namespace studio;
 
 
+Widget_Link::Widget_Link():
+	ObjectBase            ("widget_link"),
+	IS_WIDGET_FROM_BUILDER(true),
+	value_active          (*this, "tooltip_active"  , ""),
+	value_inactive        (*this, "tooltip_inactive", "")
+{
+	init();
+}
 
-namespace studio {
+Widget_Link::Widget_Link(const string &tlt_inactive, const string &tlt_active):
+	ObjectBase            ("widget_link"),
+	IS_WIDGET_FROM_BUILDER(false),
+	value_active          (*this, "tooltip_active"  , tlt_active),
+	value_inactive        (*this, "tooltip_inactive", tlt_inactive)
+{
+	init(tlt_inactive, tlt_active);
+}
 
-Widget_Link::Widget_Link(const std::string &tlt_inactive, const std::string &tlt_active)
+Widget_Link::Widget_Link(BaseObjectType *cobject):
+	ObjectBase            ("widget_link"),
+	ToggleButton          (cobject),
+	IS_WIDGET_FROM_BUILDER(true),
+	value_active          (*this, "tooltip_active"  , ""),
+	value_inactive        (*this, "tooltip_inactive", "")
 {
-	const Glib::RefPtr<Gtk::StyleContext> context = get_style_context();
+	init();
+}
 
-	// hardfixed icon size. chain icon is not a square but a rectangle.
-	Glib::RefPtr<Gtk::IconSet> chain_icon = Gtk::IconSet::lookup_default(Gtk::StockID("synfig-utils_chain_link_off"));
-	Glib::RefPtr<Gdk::Pixbuf> chain_icon_pixbuff = chain_icon->render_icon_pixbuf(context, (Gtk::IconSize)-1);
-	Glib::RefPtr<Gdk::Pixbuf> chain_icon_pixbuff_scaled = chain_icon_pixbuff->scale_simple(16, 32, Gdk::INTERP_BILINEAR);
-	// not use manage() otherwise the not-shown icon at exit wouldn't be deleted...
-	icon_off_ = new Gtk::Image(chain_icon_pixbuff_scaled);
+Widget_Link::Widget_Link(
+		BaseObjectType *cobject, const RefPtr<Builder> &builder,
+		const string tlt_inactive, const string tlt_active):
+	ObjectBase            ("widget_link"),
+	ToggleButton          (cobject),
+	builder               (builder),
+	IS_WIDGET_FROM_BUILDER(true),
+	value_active          (*this, "tooltip_active"  , tlt_active),
+	value_inactive        (*this, "tooltip_inactive", tlt_inactive)
+{
+	init();
+}
 
-	chain_icon = Gtk::IconSet::lookup_default(Gtk::StockID("synfig-utils_chain_link_on"));
-	chain_icon_pixbuff_scaled = chain_icon->render_icon_pixbuf(context, (Gtk::IconSize)-1)->scale_simple(16, 32, Gdk::INTERP_BILINEAR);
-	// not use manage() otherwise the not-shown icon at exit wouldn't be deleted...
-	icon_on_ = new Gtk::Image(chain_icon_pixbuff_scaled);
+void
+Widget_Link::init()
+{
+	set_up_icons();
 
-	icon_off_->set_padding(0,0);
-	icon_on_->set_padding(0,0);
+	refresh_tooltip_text();
+}
 
-	icon_off_->show();
-	add(*icon_off_);
-	set_relief(Gtk::RELIEF_NONE);
+void
+Widget_Link::init(const string &tlt_inactive, const string &tlt_active)
+{
+	set_up_icons();
 
 	tooltip_inactive_ = tlt_inactive;
-	tooltip_active_ = tlt_active;
+	tooltip_active_   = tlt_active;
+
 	set_tooltip_text(tooltip_inactive_);
 }
 
-Widget_Link::~Widget_Link() {
-	delete icon_on_;
-	delete icon_off_;
-}
```

Comment:
Why did you remove the destructor?
It should avoid the memory leak of those icons.

Reply:
I don't think I explicitly removed that destructor. I still have them intact on my local system here.
Wait let me squash all those commits together.

Reply:
Found it! Line 187-190

Maybe git did that on purpose because more lines were added.